hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -296,6 +302,11 @@ private void setHmsTableParameters(String newMetadataLocation, Table tbl, Map<St
       parameters.remove(hive_metastoreConstants.META_TABLE_STORAGE);
     }
 
+    // Set the basic statistics
+    parameters.put(StatsSetupConst.NUM_FILES, summary.getOrDefault(SnapshotSummary.TOTAL_DATA_FILES_PROP, "0"));
+    parameters.put(StatsSetupConst.ROW_COUNT, summary.getOrDefault(SnapshotSummary.TOTAL_RECORDS_PROP, "0"));
+    parameters.put(StatsSetupConst.TOTAL_SIZE, summary.getOrDefault(SnapshotSummary.TOTAL_FILE_SIZE_PROP, "0"));
```

Comment:
If the summary property is missing, should we set the Hive property? I think this is correct only if `"0"` indicates to Hive that the value is not known.

Reply:
Good question.
Based on [this](https://github.com/apache/hive/blob/a0034284fe02a5012f883704fcd57652519a4cd5/ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStats.java#L238-L250) Hive considers `0` as a non valid statistics value anyway:
```
  public BasicStats(Partish p) {
    partish = p;

    rowCount = parseLong(StatsSetupConst.ROW_COUNT);
[..]
    currentNumRows = rowCount;
[..]
    if (currentNumRows > 0) {
      state = State.COMPLETE;
    } else {
      state = State.NONE;
    }
  }
```

But I agree that unsetting the value would be more intuitive. What is the content of the `SnapshotSummary` in case of a new table? Setting `0` there would still be good (even if Hive does not consider it as a valid one)

Reply:
Indeed, with `"0"`, Hive treats it as unknown. It checks whether this value is `<= 0` and if so, then launches the stat collection task.

Reply:
When creating a table, `metadata.currentSnapshot()` is null, therefore we won't have a summary and will create an empty map for it. I think @rdblue is right that we should only update the HMS values if they're actually present in the summary.

Reply:
Should we remove them if they are not present for whatever reasons?

What happens if a commit is a metadata only change? Do we still have the summary?

Thanks,
Peter

Reply:
I checked with a `PropertiesUpdate`, and it does not produce a new snapshot, so `metadata.currentSnapshot()` will be the same as before along with its summary object.